UnitTesting-master/src/test/java/seleniumMail/Pages/SeleniumEasyTableDemoPage.java

```diff
@@ -0,0 +1,72 @@
+package seleniumMail.Pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
+import seleniumMail.Helpers.User;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SeleniumEasyTableDemoPage {
+
+    private static By numberOfEntriesDdl = By.cssSelector("[name='example_length']");
+    private static By tableRow = By.xpath("//tbody//tr");
+    private static By nextPaginationButtons = By.xpath("//a[@class= 'paginate_button ']");
+
+    private static WebDriver driver;
+
+    public SeleniumEasyTableDemoPage(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public SeleniumEasyTableDemoPage setEntriesNumberToShow(int entriesNumber) {
+        Select numberOfEntriesSelector = new Select(driver.findElement(numberOfEntriesDdl));
+        numberOfEntriesSelector.selectByVisibleText(String.valueOf(entriesNumber));
+        return this;
+    }
+
+    public int getNumberOfRows() {
+        List<WebElement> tableRows = driver.findElements(tableRow);
+        return  tableRows.size();
+    }
+
+    public List<User> getCustomObjects(int startAge, int maxSalary) {
```

Comment:
Update code with help of the following information: findElement method can be executed on WebElement instance. In this case new element will be found inside current. F.e. (pseudo-code)
```
WebElement headerBlock = driver.findElement(By.xpath("//div[@class='header']"));
WebElement logoutLink = headerBlock.findElement(By.xpath(".//a")); //do not forget about dot for xpath in these cases
```

Reply:
You don't need indexes for rows. Example:
`tableRow.findElement(By.xpath(".//td[1]"))`